analysis/shamanenhancement/src/modules/talents/HotHand.tsx

```diff
@@ -21,19 +23,63 @@ const HOT_HAND = {
  *
  */
 class HotHand extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+    haste: Haste,
+  };
+  protected spellUsable!: SpellUsable;
+  protected haste!: Haste;
+
   protected buffedLavaLashDamage: number = 0;
 
   constructor(options: Options) {
     super(options);
 
     this.active = this.selectedCombatant.hasTalent(SPELLS.HOT_HAND_TALENT.id);
 
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.HOT_HAND_BUFF),
+      this.reduceLavaLushCDOnBuff,
+    );
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.LAVA_LASH),
+      this.reduceLavaLushCDOnCast,
+    );
+
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.HOT_HAND_BUFF),
+      this.restoreLavaLushCDRemoveBuff,
+    );
+
     this.addEventListener(
       Events.damage.by(SELECTED_PLAYER).spell(SPELLS.LAVA_LASH),
       this.onLavaLashDamage,
     );
   }
 
+  calcLavaLushCd75pct() {
+    return (9 / (1 + this.haste.current)) * 1000;
+  }
+
+  reduceLavaLushCDOnBuff() {
+    if (this.spellUsable.isOnCooldown(SPELLS.LAVA_LASH.id)) {
+      this.spellUsable.reduceCooldown(SPELLS.LAVA_LASH.id, this.calcLavaLushCd75pct());
+    }
+  }
+
+  reduceLavaLushCDOnCast() {
+    if (this.selectedCombatant.hasBuff(SPELLS.HOT_HAND_BUFF.id)) {
+      this.reduceLavaLushCDOnBuff();
+    }
+  }
+
+  restoreLavaLushCDRemoveBuff() {
+    if (this.spellUsable.isOnCooldown(SPELLS.LAVA_LASH.id)) {
+      this.spellUsable.cooldownRemaining(SPELLS.LAVA_LASH.id);
```

Comment:
this won't modify the cooldown remaining, unless there is some very strange implicit behavior going on behind the scenes.

Reply:
Oh no, this code is obsolete. I have removed it.